### PR TITLE
Allow sorting by first execution time

### PR DESCRIPTION
**Pull request by IanButterworth**
Adds feature in #124 to allow sorting by the first execution time

```
julia> to = TimerOutput()

julia> @timeit to "c" sleep(0.1)

julia> @timeit to "c" sleep(0.1)

julia> @timeit to "c" sleep(0.1)

julia> @timeit to "b" sleep(0.1)

julia> @timeit to "a" sleep(0.1)

julia> @timeit to "c" sleep(0.1)

julia> show(to, sortby = :firstexec)
 ──────────────────────────────────────────────────────────────────
                           Time                   Allocations      
                   ──────────────────────   ───────────────────────
 Tot / % measured:      94.5s / 0.66%           50.8MiB / 0.00%    

 Section   ncalls     time   %tot     avg     alloc   %tot      avg
 ──────────────────────────────────────────────────────────────────
 c              4    417ms  66.7%   104ms   1.25KiB  66.7%     320B
 b              1    104ms  16.7%   104ms      320B  16.7%     320B
 a              1    104ms  16.6%   104ms      320B  16.7%     320B
 ──────────────────────────────────────────────────────────────────
```

### CODE REVIEW

File: src/TimerOutput.jl [hunk 1]
```diff
@@ -5,15 +5,17 @@ mutable struct TimeData
     ncalls::Int
     time::Int64
     allocs::Int64
+    firstexec::Int64
 end
-
+TimeData(ncalls, time, allocs) = TimeData(ncalls, time, allocs, time)
```
Comment (IanButterworth) [inline src/TimerOutput.jl:10]: Perhaps this needs typedefs? I didn't add them as `time_ns()` returns `UInt64` but time is `Int64` so wasn't sure what to do

**Review by KristofferC — COMMENTED** [review]
Looks ok but could use a test.

File: src/TimerOutput.jl [hunk 2]
```diff
@@ -5,15 +5,17 @@ mutable struct TimeData
...
 Base.copy(td::TimeData) = TimeData(td.ncalls, td.time, td.allocs)
 TimeData() = TimeData(0, 0, 0)
```
Comment (Octogonapus) [inline src/TimerOutput.jl:12]: To fix the tests, do this
```suggestion
TimeData() = TimeData(0, 0, 0, time_ns())
```

Comment (IanButterworth) [reply src/TimerOutput.jl]: Thanks!

**Pull request merged**